iceoryx2-bb/elementary/src/package_version.rs

```diff
@@ -63,26 +62,11 @@ impl PackageVersion {
 
     /// Returns the current [`PackageVersion`]
     pub fn get() -> PackageVersion {
-        static PACKAGE_VERSION: IoxAtomicU64 = IoxAtomicU64::new(0);
+        const MAJOR: u16 = 0;
+        const MINOR: u16 = 4;
+        const PATCH: u16 = 1;
 
-        if PACKAGE_VERSION.load(Ordering::Relaxed) == 0 {
-            let major = option_env!("CARGO_PKG_VERSION_MAJOR")
-                .and_then(|s| s.parse::<u16>().ok())
-                .unwrap_or(u16::MAX);
-            let minor = option_env!("CARGO_PKG_VERSION_MINOR")
-                .and_then(|s| s.parse::<u16>().ok())
-                .unwrap_or(u16::MAX);
-            let patch = option_env!("CARGO_PKG_VERSION_PATCH")
-                .and_then(|s| s.parse::<u16>().ok())
-                .unwrap_or(u16::MAX);
-
-            PACKAGE_VERSION.store(
-                PackageVersion::from_version(major, minor, patch).0,
-                Ordering::Relaxed,
-            );
-        }
-
-        PackageVersion::from_u64(PACKAGE_VERSION.load(Ordering::Relaxed))
+        PackageVersion::from_version(MAJOR, MINOR, PATCH)
```

Comment:
just curious, do we have a specific version convention? not sure why we didn't use the semver, which is popular in rust. so in the future we may have `beta`, `alpha` or `rc` versions.

https://crates.io/crates/semver

Reply:
In general we try to be very cautious to introduce external crates. For any other project I would agree with you but iceoryx2 shall be certified as ASIL D according to ISO26262 which is the highest safety standard in the automotive world.

This requires 100% MC/DC coverage for every line of code. Every test must be covered by a requirement, we need detailed architecture documents and documentation, a safety manual and so on.

So, every dependency we introduce must satisfy all of this! No exception. But actually, nearly none of them does it, and it is a challenge to do it. We have a strategy in place how to solve this but one of the key factors is - try to avoid external crates as much as possible.

Another answer we need to find is, how to handle or prevent supply chain attacks.

So in light of all of this, we try to be very cautious with external dependencies since the more we have the more work we have to do in the near future.

Reply:
>  ASIL D according to ISO26262 ... how to handle or prevent supply chain attacks.

understood, quite a bit different from the normal project. thanks for your explaination:)